doc/release/1.7.3-notes.rst

```diff
@@ -0,0 +1,37 @@
+==========================
+SciPy 1.7.3 Release Notes
+==========================
+
+.. contents::
+
+SciPy 1.7.3 is a bug-fix release that provides binary wheels
+for MacOS arm64 with Python 3.8, 3.9, and 3.10. Note that a small
```

Comment:
I would not mention that tests were skipped - we skip tests on all releases. If there are serious enough problems that they deserve pointing out, then we should just be explicit - "feature X may have issues, see xxx"